websites/I/IFTTT/presence.ts

```diff
@@ -8,270 +8,357 @@ const presence = new Presence({
-            }
-          }
-          break;
-        //My services
-        case "date_and_time":
-        case "email":
-        case "email_digest":
-        case "ifttt":
-        case "feed":
-        case "space":
-        case "weather":
-        case "maker_webhooks":
-          presenceData.details = "My Services";
-          presenceData.state = document.querySelector("h1").textContent;
-          break;
-        case "my_services":
-          presenceData.details = "My Services";
-          break;
-        //Connect services
-        case "connect":
-          presenceData.details = "My Services";
-          presenceData.state = `Connect ${
-            (document.querySelector("h1").firstElementChild as HTMLTitleElement)
-              .title
-          } to ${
-            (document.querySelector("h1").lastElementChild as HTMLTitleElement)
-              .title
-          }`;
-          break;
-        //Explore
-        case "discover":
-          presenceData.details = "Explore";
-          break;
-        //Search
-        case "search":
-          presenceData.smallImageKey = "search";
-          presenceData.smallImageText = (await strings).search;
+/**
+ * Handle blog pages
+ * @param entry Blog entry name
+ */
+async function handleBlog(entry?: string): Promise<void> {
+  presence.setActivity({
+    details: "Blog",
+    state: entry,
+    largeImageKey: "logo_big"
+  });
+}
 
-          presenceData.details = `Searching for "${
-            (document.getElementById("search") as HTMLInputElement).value
-          }"`;
-          presenceData.state = `Tab: ${
-            document.getElementsByClassName("active")[1].textContent
-          }`;
-          break;
-        //Trust & Privacy
-        case "terms":
-          presenceData.details = "Privacy Policy & Terms of Use";
-          break;
-        //Careers
-        case "careers":
-          presenceData.details = "Careers";
-          break;
-        //Unknown
-        default:
-          if (document.getElementsByClassName("brand-section").length > 0) {
-            presenceData.details = "Services";
-            presenceData.state = document.querySelector("h1").textContent;
-          } else {
-            presence.setActivity();
-            return;
-          }
-      }
-      break;
-  }
+/**
+ * Handle my services pages
+ * @param category Service category
+ */
+async function handleMyServices(category?: string): Promise<void> {
+  presence.setActivity({
+    details: "My Services",
+    state: category,
+    largeImageKey: "logo_big",
+    smallImageText: (await strings).browsing,
+    smallImageKey: "reading"
+  });
+}
 
-  presence.setActivity(presenceData);
-});
+/**
+ * Handle custom pages
+ * @param details Presence details text
+ * @param state Presence state text
+ */
+async function handleCustom(details: string, state?: string): Promise<void> {
+  presence.setActivity({
+    details,
+    state,
+    largeImageKey: "logo_big"
+  });
+}
```

Comment:
Can't you just use `presenceData.details` and `presenceData.state` instead? 
Seems very useless as half the time you just pass details

Reply:
I'd have to write these three lines over and over again. This function is just to make it a one-liner and therefor more readable. If this is a problem and needs to be changed, then I'm going to change it again.

Reply:
Just repeat it

Reply:
Alright, I'm going to write it 12 times then.